File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    
    def cut(s, start):
        count1 = 1
        count2 = 0
        target = s[start]
        for i in range(start+1, len(s)):
            if s[i] == target:
                count1 += 1
            else:
                count2 += 1
            if count1 == count2:
                return i
        return len(s)-1
    
    start = end = 0
    while True:
        if start == len(s):
            break
        end = cut(s, start)
        answer += 1
        start = end + 1
        
    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
import java.util.*;
class Solution {
    public int solution(String s) {
        int answer = 0;
        int start = 0;
        int end = 0;
        while(true){
            if(start == s.length()){
                break;
            }
            end = cut(s, start);
            answer++;
            start = end + 1;
        }
        return answer;
    }
    
    public int cut(String s, int start){
        char ch = s.charAt(start);
        int count1 = 1;
        int count2 = 0;
        for(int i=start+1; i<s.length(); i++){
            if(s.charAt(i) == ch){
                count1++;
            } else {
                count2++;
            }
            if(count1 == count2){
                return i;
            }
        }
        return s.length()-1;
    }
}
